javaparser-core/src/main/java/com/github/javaparser/utils/ProjectRoot.java

```diff
@@ -0,0 +1,46 @@
+package com.github.javaparser.utils;
+
+import com.github.javaparser.ParserConfiguration;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * A collection of SourceRoots. The ProjectRoot can be provided and populated with a CollectionContext together with
+ * a CollectionStrategy.
+ */
+public class ProjectRoot {
```

Comment:
This concept makes sense to me but I am not sure the Javadoc is 100% clear.

What means that the ProjectRoot can be provided?
Why I need to interact both with a CollectionContext and a CollectionStrategy?

Reply:
Maybe this needs an article like http://javaparser.org/setting-java-8-9-10-etc/ @daanschipper if that's too much for you, a bit of sample code will do and I'll write it :-)

Reply:
I've chosen to implement a strategy pattern for this since different files (for instance .jars) need to be collected to construct the ProjectRoot. The ProjectRoot is basically a list of SourceRoots, which are all set with a ParserConfiguration. For just parsing the compilation units, you only need the SourceRoots. However, if you also want to resolve expressions within the compilation unit, you also need to initialise a TypeSolver. This requires a different strategy which you can set with the CollectionContext.